openedx/core/djangoapps/notifications/management/commands/update_notification_preferences.py

```diff
@@ -0,0 +1,116 @@
+"""
+Management command for updating notification preferences with parameters
+"""
+import logging
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+
+from openedx.core.djangoapps.notifications.models import NotificationPreference
+from openedx.core.djangoapps.notifications.base_notification import (
+    COURSE_NOTIFICATION_APPS,
+    COURSE_NOTIFICATION_TYPES
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Management command to update boolean notification preferences.
+
+    This command updates channel (`web`, `push`, `email`)
+    in the NotificationPreference model for a given app and type.
+
+    Features:
+        - Requires `app` and `type`, validated against
+          COURSE_NOTIFICATION_APPS and COURSE_NOTIFICATION_TYPES.
+        - Allows updating a single channel to `true` or `false`.
+        - Supports optional `--user_ids` argument to limit updates
+          to specific users.
+        - Provides a `--dry-run` mode to preview changes without
+          committing to the database.
+        - Logs the number of affected records and affected user IDs.
+
+    Example usage:
+        python manage.py update_notification_preference discussion new_comment_on_response email false
+        python manage.py update_notification_preference discussion new_comment_on_response push false --user_ids 5 7 12
+        python manage.py update_notification_preference discussion new_comment_on_response web false --dry-run
+    """
+    help = "Update boolean notification preferences for users at account level."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "app",
+            type=str,
+            choices=list(COURSE_NOTIFICATION_APPS.keys()),
+            help=f"App key (choices: {', '.join(COURSE_NOTIFICATION_APPS.keys())})",
+        )
+        parser.add_argument(
+            "type",
+            type=str,
+            choices=list(COURSE_NOTIFICATION_TYPES.keys()),
+            help=f"Type key (choices: {', '.join(COURSE_NOTIFICATION_TYPES.keys())})"
+        )
```

Comment:
We shoudl make `app` and/or `notification_type` required  or change the queryset to handle this case. 

`queryset = NotificationPreference.objects.filter(app=app, type=pref_type)`

If `app` and `notification_type` is not provided, it will be empty queryset

Reply:
They are required by default because they’re positional args.